### PR TITLE
run-make-check.sh: Adding ccache tuning for the CI

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
When run-make-check is run by the CI, some tuning can be performed to
speedup the build.

This commit :
- Detect jenkins by searching JENKINS_HOME env variable
- Defines the SOURCE_DATE_EPOCH to enforce a stable date across builds
- Ask cmake not to use the git versioning which adds useless entropy for a temporary build (ENABLE_GIT_VERSION=OFF)
- Define the ccache slopiness to increase efficiency
- Increase the ccache size to save multiple builds to maximise cache hit between PRs
- Print ccache statistics to evaluate ccache efficiency

Signed-off-by: Erwan Velu <erwan@redhat.com>

### CODE REVIEW

**Comment by ErwanAliasr1** [comment]
That would be also worth backporting it to luminous as some PRs could be speedup also.

File: run-make-check.sh [hunk 1]
```diff
@@ -74,9 +74,30 @@ function run() {
...
-    $DRY_RUN ./do_cmake.sh -DWITH_FIO=ON -DWITH_GTEST_PARALLEL=ON $CMAKE_PYTHON_OPTS $@ || return 1
+    CMAKE_BUILD_OPTS="-DWITH_FIO=ON -DWITH_GTEST_PARALLEL=ON"
+
+    # Are we in the CI ?
+    if [ -n "$JENKINS_HOME" ]; then
+      echo "Jenkins got detected, let's tune the build"
+      # The following settings are made for improving ccache efficiency
+      # by removing the entropy generated by the date/time embedded in the build
+      CMAKE_BUILD_OPTS="$CMAKE_BUILD_OPTS -D ENABLE_GIT_VERSION=OFF"
+      export SOURCE_DATE_EPOCH=$(date +%D |date -f- +%s);
```
Comment (leseb) [inline run-make-check.sh]: Why `;` at the end of line?

File: run-make-check.sh [hunk 2]
```diff
@@ -74,9 +74,30 @@ function run() {
...
+      ccache -o sloppiness=time_macros
+      ccache -o run_second_cpp=true
+
+      # Build host have plenty of space available, let's use it to keep
```
Comment (leseb) [inline run-make-check.sh]: s/have/has/

File: run-make-check.sh [hunk 3]
```diff
@@ -74,9 +74,30 @@ function run() {
...
+      # various versions of tthe built objects. This could increase the cache hit
```
Comment (leseb) [inline run-make-check.sh]: s/tthe/the/

**Comment by ErwanAliasr1** [comment]
@leseb thanks ! Fixed.

**Comment by ErwanAliasr1** [comment]
@tchaikov can we backport that to luminous to have also the benefit of it ?

**Comment by tchaikov** [comment]
@smithfarm what do you think? guess we could have this change in mimic and luminous.

**Pull request merged**